presto-raptorx/src/main/java/com/facebook/presto/raptorx/storage/OrcFileWriter.java

```diff
@@ -0,0 +1,144 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx.storage;
+
+import com.facebook.presto.orc.OrcWriter;
+import com.facebook.presto.orc.OrcWriterStats;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.PageBuilder;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.raptorx.RaptorErrorCode.RAPTOR_STORAGE_ERROR;
+import static com.facebook.presto.raptorx.util.OrcUtil.createOrcFileWriter;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class OrcFileWriter
+        implements Closeable
+{
+    private final PageBuilder pageBuilder;
+    private final OrcWriter orcWriter;
+
+    private boolean closed;
+    private long rowCount;
+    private long uncompressedSize;
+
+    public OrcFileWriter(
+            long chunkId,
+            List<Long> columnIds,
+            List<Type> columnTypes,
+            CompressionType compressionType,
+            File target,
+            TypeManager typeManager,
+            OrcWriterStats stats)
+    {
+        StorageTypeConverter converter = new StorageTypeConverter(typeManager);
+        List<Type> storageTypes = columnTypes.stream()
+                .map(converter::toStorageType)
+                .collect(toImmutableList());
+
+        this.pageBuilder = new PageBuilder(storageTypes);
+
+        Map<String, String> metadata = OrcFileMetadata.from(chunkId, columnIds, columnTypes).toMap();
+        this.orcWriter = createOrcFileWriter(target, columnIds, storageTypes, compressionType, metadata, stats);
+    }
+
+    public void appendPages(List<Page> pages)
+    {
+        for (Page page : pages) {
+            appendPage(page);
+        }
+    }
+
+    public void appendPages(List<Page> pages, int[] pageIndexes, int[] positionIndexes)
+    {
+        checkArgument(pageIndexes.length == positionIndexes.length, "pageIndexes and positionIndexes do not match");
+
+        for (int i = 0; i < pageIndexes.length; i++) {
+            Page page = pages.get(pageIndexes[i]);
+            int position = positionIndexes[i];
+            appendPositionTo(page, position, pageBuilder);
+
+            if (pageBuilder.isFull()) {
+                appendPage(pageBuilder.build());
```

Comment:
"highker on Feb 4  Member
This confuses with appendPages... Maybe call flush or write"

Reply:
"highker on Feb 4  Member
I think ORC writer has its own buffer so we don't need to flush until we reach a full page. Directly call Writer::write shouldn't harm"

Reply:
You are very right, a lot of things in this file is confusing. 

As I summarized in the "CodeReview" doc, we need think the whole writing path together, and do another task,  after that, maybe even OrcFileWriter.java doesn't exist, From PageBuffer directly go to OrcFileWriter.